skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -175,46 +171,140 @@ def transform(self, K, y=None):
-            - np.broadcast_arrays(K, self.ref_cmean)[1]
+            - np.broadcast_arrays(K, self.K_fit_rows_)[1]
             - rmean.reshape((K.shape[0], 1))
-            + np.broadcast_arrays(K, self.ref_mean)[1]
-        )
+            + np.broadcast_arrays(K, self.K_fit_all_)[1]
+        ) / self.scale_
 
         return Kc
 
-    def fit_transform(self, K, y=None, ref_cmean=None, ref_mean=None, **fit_params):
+    def fit_transform(self, K, y=None, K_fit_rows=None, K_fit_all=None, **fit_params):
         r"""Fit to data, then transform it.
 
         :param K: Kernel matrix
         :type K: ndarray of shape (n_samples, n_samples)
         :param y: ignored
-        :param ref_cmean: an array with means for each column.
-        :type ref_cmean: array of shape (1, n_features)
-        :param ref_mean: an average for the whole kernel matrix
-        :type ref_mean: array
+        :param K_fit_rows: an array with means for each column.
+        :type K_fit_rows: array of shape (1, n_features)
+        :param K_fit_all: an average for the whole kernel matrix
+        :type K_fit_all: array
         :param \**fit_params: necessary for compatibility with the functions of the TransformerMixin class
 
         :return: tranformed matrix Kc
         """
-        self.fit(K, y, ref_cmean=ref_cmean, ref_mean=ref_mean)
+        self.fit(K, y, K_fit_rows=K_fit_rows, K_fit_all=K_fit_all)
         return self.transform(K, y)
+
+
+class SparseKernelCenterer(TransformerMixin, BaseEstimator):
+    """Kernel centering method for sparse kernels, similar to
+    KernelFlexibleCenterer
+    """
+
+    def __init__(self, rcond=1e-12):
+        """
+        Initialize SparseKernelCenterer.
+
+        :param rcond: conditioning parameter to use when computing the
+                      Nystrom-approximated kernel for scaling
+        :type rcond: float, default 1E-12
+        """
+
+        self.rcond = rcond
+
+    def fit(self, Knm, Kmm, y=None):
+        """Fit KernelFlexibleCenterer
+
+        :param Knm: Kernel matrix between the reference data set and the active
+                    set
+        :type Knm: ndarray of shape (n_samples, n_active)
+
+        :param Kmm: Kernel matrix between the active set and itself
+        :type Kmm: ndarray of shape (n_active, n_active)
+
+        :param y: ignored
+
+        :return: itself
+        """
+
+        if Knm.shape[1] != Kmm.shape[0]:
+            raise ValueError(
+                "The reference kernel is not commensurate shape with the"
+                "active kernel."
+            )
+
+        if Kmm.shape[0] != Kmm.shape[1]:
+            raise ValueError("The active kernel is not square.")
+
+        self.n_active_ = Kmm.shape[0]
+
+        self.K_fit_rows_ = Knm.mean(axis=0)
+
+        Knm_centered = Knm - np.broadcast_arrays(Knm, self.K_fit_rows_)[1]
+
+        Khat = Knm_centered @ np.linalg.pinv(Kmm, self.rcond) @ Knm_centered.T
+
+        self.scale_ = np.sqrt(np.trace(Khat) / Knm.shape[0])
```

Comment:
Can we make the centering and the scaling separate, i.e., add the capability to do one without the other?